src/rust/engine/process_execution/src/immutable_inputs.rs

```diff
@@ -42,20 +46,41 @@ impl ImmutableInputs {
     let value: Result<_, String> = cell
       .get_or_try_init(async {
```

Comment:
So ... we pass a future and not a closure! The API is wrong. Even though the future is only awaited once other futures can get spawned, never awaited, but still make progress.

Reply:
Untrue.